hnn_core/network.py

```diff
@@ -160,9 +160,10 @@ def __init__(self, params):
         # Number of time points
         # Originally used to create the empty vec for synaptic currents,
         # ensuring that they exist on this node irrespective of whether
-        # or not cells of relevant type actually do
-        self.n_times = np.arange(0., self.params['tstop'],
-                                 self.params['dt']).size + 1
+        # or not cells of relevant type actually
+        self.t_vec = np.arange(0., params['tstop'] + params['dt'],
+                               params['dt'])
```

Comment:
can we call `t_vec` to be `times`? It's more descriptive. Also is `n_times` attribute really needed? If not, let's remove it. Also update the docstring!

Reply:
Renaming to `times` should be fine as far as I can tell. More consistent too since `t_vec` is passed to the `times` attribute of `Dipole`.

`n_times` is never used explicitly, but it is referenced in the doc string of every example, as well as the `Dipole` class. Obviously someone thought it was important at some point, but it's pretty trivial remove if you still want to.

Reply:
I spoke too soon. `times` is an attribute of the `Spikes` class. Perhaps we can rename that to `spiketimes`?

Reply:
yep, let's rename that one to `spike_times` and we can remove `n_times` if it's not used anywhere

Reply:
I think we're verging into a problem caused by conflating two classes that shouldn't be combined. `Spikes` was originally designed to represent instances of spikes. As such, it has corresponding attributes of `Spikes.times`, `Spikes.gids`, and `Spikes.types`. In order to provide a clear and consistent API, I think there are a few options we should consider:

1. Separate the `Spikes` class from the `CellResponse` class
2. If we want to rename `Spikes.times` to `Spikes.spiketimes`, I think we should rename all of the other spike-related attributes to be consistent (e.g., also rename the `gids` attribute to `spikegids`)
3. Rename `Spikes.times` to something like `Spikes.time_trajectory` to indicate that the time values specified in this attribute include all time steps rather than just the ones that pertain to spikes

Reply:
Either 2. or 3. is fine for me. Maybe slight preference for 2. since `times` is used elsewhere in `Dipole` object etc. with the same meaning, so you'd have to change it there too if you went with 3.

Reply:
I agree, my preference is for 2 as well. Given how closely coupled spikes, currents, and voltages are I think it is best to keep them grouped and rename `Spikes` to `CellResponse` either in this PR, or in #199.

I've been putting off renaming since just adding voltages doesn't seem enough to justify it yet. Given the confusion @rythorpe is bringing up, do you think it should be done in this PR?

Reply:
Maybe let's save it for #199 so that we can move forward with this PR.